### PR TITLE
Correction of assert in case a 32bits architecture is used

### PR DESCRIPTION
**Pull request by jmgc**
There is an assert that checks the size of a type against the pointer size. This does not work on 32 bits architectures as the pointers are smaller, but the other types are not reduced acordingly.

### CODE REVIEW

File: lib/ClangImporter/SwiftLookupTable.h [hunk 1]
```diff
@@ -133,8 +133,14 @@ class EffectiveClangContext {
     return StringRef(Unresolved.Data, UnresolvedLength);
   }
 };
+
+#if __SIZEOF_POINTER__ == 4
```
Comment (CodaFi) [inline lib/ClangImporter/SwiftLookupTable.h]: Is this defined on MS platforms? @hughbe @compnerd

**Comment by CodaFi** [comment]
Could you please rebase and drop these merge commits?

File: lib/ClangImporter/SwiftLookupTable.h (hunk 1 above)
Comment (hughbe) [inline lib/ClangImporter/SwiftLookupTable.h]: Nope. __SIZEOF_POINTER__ is undefined 

Use LLVM_PTR_SIZE defined in "LLVM/Support/Compiler.h"

https://github.com/llvm-mirror/llvm/blob/master/include/llvm/Support/Compiler.h

**Review by hughbe — CHANGES_REQUESTED** [review]

**Comment by jmgc** [comment]
Ok, I close this pull request and ask for a new one with a rebased version and using SIZEOF_POINTER.

**Pull request closed without merging**